.github/workflows/sphinx.yml

```diff
@@ -0,0 +1,51 @@
+name: Build Documentation
+
+on:
+  push:
+    branches:
+      - 'main'
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+jobs:
+  build:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+      - name: Set up Python
+        id: setup_python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+      - name: Cache pip
+        uses: actions/cache@v3
+        id: pip-cache
+        with:
+          key: ${{ runner.os }}-${{ env.pythonLocation }}-pip-${{ hashFiles('**/requirements.txt') }}
+          path: ${{ env.pythonLocation }}
+        if: steps.pip-cache.outputs.cache-hit != 'true'
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt
+
+      - name: Build html
+        run: |
+          sphinx-apidoc -o doc src
+          cd doc/
```

Comment:
The `doc` folder is not empty. Maybe best to create an empty folder to use for the sphinx docs?

Reply:
I used it on the existing folder in case we want to integrate the existing files in `doc` into the website, but if this is not the case I'll create another folder for sphinx.

Reply:
Mostly that there are two build steps and it's putting intermediate files into the doc folder as well as the final files? Maybe there's no real intermediate, I can't check at the moment.

But it looks like those other docs are more about git contribution and not usage. I'm thinking is best to keep sources of documentation separate and published separately.